lib/model.js

```diff
@@ -2977,11 +2977,12 @@ class Model {
    * @param {any} value
    */
   setDataValue(key, value) {
```

Comment:
This does't work if you call setDataValue multiple times... `previous` should return the original value (ie. the value in the DB when the record was fetched), am I wrong?